File: ios/JJMproject/Classes/JSBridge/Modules/EventEmitter/ReactNativeEventEmitter.m
//
//  ReactNativeEventEmitter.m
//  MarioFramework
//
//  Created by MeePwn on 2017/11/27.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ReactNativeEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/JJMproject/Classes/JSBridge/Modules/EventEmitter/ReactNativeEventEmitter.swift
//
//  ReactNativeEventEmitter.swift
//  MarioFramework
//
//  Created by MeePwn on 2017/11/27.
//  Copyright © 2017年 Facebook. All rights reserved.
//

import Foundation

@objc(ReactNativeEventEmitter)
open class ReactNativeEventEmitter: RCTEventEmitter {
  
  override init() {
    super.init()
    EventEmitter.sharedInstance.registerEventEmitter(eventEmitter: self)
  }
  
  @objc open override func supportedEvents() -> [String] {
    return EventEmitter.sharedInstance.allEvents
  }
  
}
